### DISCUSSION TITLE
error when execution instruction catkin_make -DUSE_PANGOLIN_VIEWER=ON -DUSE_SOCKET_PUBLISHER=OFF （in Build Instructions）

### DISCUSSION THREAD

**Discussion by psydbd** [stella-cv/stella_vslam#492] [Q&A] [answered]
These are all the situations.

root@lyy-vpc:~/catkin_ws# catkin_make -DUSE_PANGOLIN_VIEWER=ON -DUSE_SOCKET_PUBLISHER=OFF
Base path: /root/catkin_ws
Source space: /root/catkin_ws/src
Build space: /root/catkin_ws/build
Devel space: /root/catkin_ws/devel
Install space: /root/catkin_ws/install
####
####
####
####
[  0%] Built target topic_tools_generate_messages_nodejs
[  0%] Built target nodelet_generate_messages_py
[  0%] Built target tf_generate_messages_cpp
[  0%] Built target tf_generate_messages_nodejs
[  0%] Built target topic_tools_generate_messages_cpp
[  0%] Built target topic_tools_generate_messages_lisp
[  0%] Built target topic_tools_generate_messages_eus
[  0%] Built target std_srvs_generate_messages_lisp
[  0%] Built target std_srvs_generate_messages_eus
[  0%] Built target pcl_msgs_generate_messages_py
[  0%] Built target std_srvs_generate_messages_cpp
[  0%] Built target pcl_msgs_generate_messages_lisp
[  0%] Built target pcl_msgs_generate_messages_nodejs
[  0%] Built target pcl_msgs_generate_messages_eus
[  0%] Built target pcl_msgs_generate_messages_cpp
[  0%] Built target bond_generate_messages_py
[  0%] Built target bond_generate_messages_nodejs
[  0%] Built target bond_generate_messages_lisp
[  0%] Built target geometry_msgs_generate_messages_eus
[  0%] Built target nav_msgs_generate_messages_eus
[  0%] Built target std_msgs_generate_messages_py
[  0%] Built target sensor_msgs_generate_messages_cpp
[  0%] Built target tf_generate_messages_eus
[  0%] Built target nav_msgs_generate_messages_nodejs
[  0%] Built target rosgraph_msgs_generate_messages_lisp
[  0%] Built target nav_msgs_generate_messages_lisp
[  0%] Built target nav_msgs_generate_messages_cpp
[  0%] Built target std_srvs_generate_messages_py
[  0%] Built target rosgraph_msgs_generate_messages_eus
[  0%] Built target pcl_ros_gencfg
[  0%] Built target geometry_msgs_generate_messages_py
[  0%] Built target tf_generate_messages_lisp
[  0%] Built target nav_msgs_generate_messages_py
[  0%] Built target sensor_msgs_generate_messages_nodejs
[  0%] Built target bond_generate_messages_eus
[  0%] Built target roscpp_generate_messages_cpp
[  0%] Built target nodelet_topic_tools_gencfg
[  0%] Built target topic_tools_generate_messages_py
[  0%] Built target sensor_msgs_generate_messages_lisp
[  0%] Built target sensor_msgs_generate_messages_eus
[  0%] Built target sensor_msgs_generate_messages_py
[  0%] Built target roscpp_generate_messages_eus
[  0%] Built target tf2_msgs_generate_messages_eus
[  0%] Built target rosgraph_msgs_generate_messages_nodejs
[  0%] Built target actionlib_msgs_generate_messages_py
[  0%] Built target actionlib_generate_messages_cpp
[  0%] Built target std_msgs_generate_messages_lisp
[  0%] Built target actionlib_msgs_generate_messages_eus
[  0%] Built target dynamic_reconfigure_generate_messages_py
[  0%] Built target roscpp_generate_messages_py
[  0%] Built target geometry_msgs_generate_messages_lisp
[  0%] Built target std_msgs_generate_messages_cpp
[  0%] Built target roscpp_generate_messages_nodejs
[  0%] Built target geometry_msgs_generate_messages_cpp
[  0%] Built target dynamic_reconfigure_generate_messages_nodejs
[  0%] Built target tf2_msgs_generate_messages_py
[  0%] Built target bond_generate_messages_cpp
[  0%] Built target rosgraph_msgs_generate_messages_py
[  0%] Built target actionlib_generate_messages_lisp
[  0%] Built target roscpp_generate_messages_lisp
[  0%] Built target actionlib_msgs_generate_messages_cpp
[  0%] Built target rosgraph_msgs_generate_messages_cpp
[  0%] Built target actionlib_generate_messages_eus
[  0%] Built target actionlib_msgs_generate_messages_lisp
[  0%] Built target std_msgs_generate_messages_eus
[  0%] Built target tf_generate_messages_py
[  0%] Built target actionlib_msgs_generate_messages_nodejs
[  0%] Built target actionlib_generate_messages_py
[  0%] Built target tf2_msgs_generate_messages_lisp
[  0%] Built target tf2_msgs_generate_messages_nodejs
[  0%] Built target dynamic_reconfigure_generate_messages_cpp
[  0%] Built target geometry_msgs_generate_messages_nodejs
[  0%] Built target actionlib_generate_messages_nodejs
[  0%] Built target nodelet_generate_messages_eus
[  0%] Built target dynamic_reconfigure_gencfg
[  0%] Built target dynamic_reconfigure_generate_messages_eus
[  0%] Built target nodelet_generate_messages_cpp
[  0%] Built target tf2_msgs_generate_messages_cpp
[  0%] Built target nodelet_generate_messages_nodejs
[  0%] Built target nodelet_generate_messages_lisp
[  0%] Built target dynamic_reconfigure_generate_messages_lisp
[  0%] Built target std_msgs_generate_messages_nodejs
[  0%] Built target std_srvs_generate_messages_nodejs
[ 33%] Linking CXX executable /root/catkin_ws/devel/lib/stella_vslam_ros/run_slam
/usr/bin/ld: warning: libopencv_calib3d.so.405, needed by /usr/local/lib/libstella_vslam.so, may conflict with libopencv_calib3d.so.4.2
/usr/bin/ld: warning: libopencv_core.so.405, needed by /usr/local/lib/libstella_vslam.so, may conflict with libopencv_core.so.4.2
/usr/bin/ld: CMakeFiles/run_slam.dir/stella_vslam_ros.cc.o: undefined reference to symbol '_ZN2cv3MatC1Ev'
/usr/bin/ld: /usr/local/lib/libopencv_core.so.405: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [stella_vslam_ros/src/CMakeFiles/run_slam.dir/build.make:288：/root/catkin_ws/devel/lib/stella_vslam_ros/run_slam] error 1
make[1]: *** [CMakeFiles/Makefile2:2827：stella_vslam_ros/src/CMakeFiles/run_slam.dir/all] error 2
make: *** [Makefile:141：all] error 2
Invoking "make -j4 -l4" failed

I don't know how to solve these two errors. I use ROS to run it. I look forward to getting help and appreciate it.

**Comment by ymd-stella** [comment, marked as answer]
Did you not read the following note?

https://stella-cv.readthedocs.io/en/latest/installation.html#:~:text=NOTE%3A%20If%20ROS%20is%20installed%2C%20use%20the%20one%20installed%20with%20ROS

> NOTE: If ROS is installed, use the one installed with ROS.

**Reply by ymd-stella** [reply]
The exact same question can be found at https://github.com/stella-cv/stella_vslam_ros/discussions/112.

**Reply by psydbd** [reply]
Thank you very much for the details, patience, timely answers.
I will start to install the Ubuntu system again.
Read the notes carefully.
You are a great master.
